src/custom-pv-components/pdb-navigation.js

```diff
@@ -17,25 +17,58 @@ const height = 40,
   };
 
 class ProtvistaPdbNavigation extends ProtvistaNavigation {
-  
+
   connectedCallback() {
     this.style.display = 'block';
     this.style.width = '100%';
     this.width = this.offsetWidth;
 
     this._offset = parseFloat(this.getAttribute('offset')) || 0;
+    this._length = parseInt(this.getAttribute('length'));
+    this._displaystart = parseInt(this.getAttribute('displaystart')) || (this._offset > 0) ? this._offset : 1;
+    this._displayend = parseInt(this.getAttribute('displayend')) || (this._offset > 0) ? (this._length + this._offset - 1) : this._length;
+    this._highlightstart = parseInt(this.getAttribute('highlightstart'));
+    this._highlightend = parseInt(this.getAttribute('highlightend'));
+    this._highlightintervals = this.getAttribute("highlightintervals");
 
-    this._length = parseFloat(this.getAttribute('length'));
-    this._displaystart = parseFloat(this.getAttribute('displaystart')) || (this._offset > 0) ? this._offset : 1;
-    this._displayend = parseFloat(this.getAttribute('displayend')) || (this._offset > 0) ? (this._length + this._offset - 1) : this._length;
-    this._highlightStart = parseFloat(this.getAttribute('highlightStart'));
-    this._highlightEnd = parseFloat(this.getAttribute('highlightEnd'));
+    this._highlightedFragments = [];
 
     this._onResize = this._onResize.bind(this);
+    this._updateNavRuler = this._updateNavRuler.bind(this);
 
     this._createNavRuler();
   }
 
+  static get observedAttributes() {
+    return ['length', 'displaystart', 'displayend', 'highlightstart', 'highlightend', 'width', 'highlightintervals'];
+  }
+
+  attributeChangedCallback(name, oldValue, newValue) {
+    if (oldValue !== newValue) {
+      if (name === "highlightintervals") this[`_${name}`] = newValue;
+      else if (name === "offset") this[`_${name}`] = parseFloat(newValue);
+      else this[`_${name}`] = parseInt(newValue);
```

Comment:
As all branches return a value to be assigned to the same place, consider: ```this[`_${name}`] = someDeclarativeFunctionName(name, newValue)```

Reply:
In this case, what about a ternary? `this[`_${name}`] = name === "highlightintervals" ? newValue : name === "offset" ? parseFloat(newValue) : parseInt(newValue);`. I mean, the function would work but as this if/else is very specific for x attributes, creating the function outside the scope of `attributeChangedCallback`.... and creating it inside I don't know if its more readable with a ternary... what do you think? 🙌🏼

Reply:
Yes, a ternary is also ok.